core/domain/exp_services.py

```diff
@@ -792,6 +793,15 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
                 'which is too old. Please reload the page and try again.'
                 % (exploration_model.version, exploration.version))
 
+    if feconf.ENABLE_STRING_CLASSIFIER:
```

Comment:
Hm. I'm afraid that's too subtle. No one's going to realize that and it's going to make the code unmaintainable; this sort of thing merits at least a comment.

That said, we should find a way to alleviate this. Why not just increment the version earlier, then save the exploration model, then handle all the classifier stuff? Basically, find a way to make what's going on as simple and elegant as possible so that it's very easy to follow.